crates/binstalk/src/fetchers.rs

```diff
@@ -44,6 +44,10 @@ pub trait Fetcher: Send + Sync {
     /// A short human-readable name or descriptor for the package source
     fn source_name(&self) -> CompactString;
 
+    /// A short human-readable name, must contains only characters
+    /// and numbers and it also must be unique.
+    fn fetcher_name(&self) -> CompactString;
```

Comment:
What's the purpose of this compared to source_name?

Reply:
`source_name` returned a url, which can include characters like ':' and '/', which I don't think is a good idea to include these characters in the directory name.

Reply:
And they are also too lengthy and confusing.
`fetcher_name` simply returns the type of the fetcher, which is short, concise and does not include special characters.

Reply:
Right, yeah. In that case, can we:
- include that it's going to be used for a dir name in the docs
- use a `&'static str` or something (or maybe can be a `const fn` with [`new_inline`](https://docs.rs/compact_str/latest/compact_str/struct.CompactString.html#method.new_inline)?)

Reply:
> - include that it's going to be used for a dir name in the docs

I would add that to the doc.

> - use a `&'static str` or something (or maybe can be a `const fn` with [`new_inline`](https://docs.rs/compact_str/latest/compact_str/struct.CompactString.html#method.new_inline)?)

Sure.

Reply:
Done!